packages/aragon-wrapper/src/index.js

```diff
@@ -197,31 +199,114 @@ export default class Aragon {
    * @return {void}
    */
   initNotifications () {
-    this.notifications = new BehaviorSubject(
-      this.cache.get('notifications', [])
-    )
-      .scan((notifications, notification) => notifications.concat(notification))
-      .do((notifications) => this.cache.set(notifications))
+    // If the cached notifications doesn't exist or isn't an array, set it to an empty one
+    let cached = this.cache.get('notifications')
+    if (!Array.isArray(cached)) {
+      cached = []
+    } else {
+      // Set up acknowledge for unread notifications
+      cached.forEach(notification => {
+        if (notification && !notification.read) {
+          notification.acknowledge = () => this.acknowledgeNotification(notification.id)
+        }
+      })
+    }
+
+    this.notifications = new BehaviorSubject(cached)
+      .scan((notifications, { modifier, notification }) => modifier(notifications, notification))
+      .do((notifications) => this.cache.set('notifications', notifications))
+      .publishReplay(1)
+    this.notifications.connect()
   }
 
   /**
    * Send a notification.
    *
+   * @param {string} app   The address of the app sending the notification
    * @param {string} title The notification title
-   * @param {string} body The notification body
+   * @param {string} body  The notification body
    * @param {object} [context={}] The application context to send back if the notification is clicked
    * @param  {Date}  [date=new Date()] The date the notification was sent
    * @return {void}
    */
   sendNotification (app, title, body, context = {}, date = new Date()) {
-    this.notifications.next({
-      title,
+    const id = uuidv4()
+    const notification = {
+      app,
       body,
       context,
-      app,
       date,
-      acknowledge: function () {
-        // TODO
+      id,
+      title,
+      read: false,
+    }
+    this.notifications.next({
+      modifier: (notifications, notification) => {
+        // Find the first notification that's not before this new one
+        // and insert ahead of it if it exists
+        const newNotificationIndex = notifications.findIndex(
+          notification => !isBefore(new Date(notification.date), date)
+        )
+        return newNotificationIndex === -1 ?
+          [...notifications, notification] :
+          [
+            ...notifications.slice(0, newNotificationIndex),
+            notification,
+            ...notifications.slice(newNotificationIndex)
+          ]
```

Comment:
The notifications are inserted sorted by date time; I thought it was better to do this work here on every insert rather than having the frontend re-sort the array on each update.